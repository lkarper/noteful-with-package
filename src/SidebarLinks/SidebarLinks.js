import React from 'react';
import { NavLink } from 'react-router-dom';
import NotesContext from '../NotesContext/NotesContext';

const SidebarLinks = () => {

    return (
        <NotesContext.Consumer>
            {value => {
                const error = value.folderError;
                const errorHTML = (
                    <div className="folder-load-error">
                        <h2>Sorry, could not load folders from the server: {error}</h2>
                        <p>Check your network connection and reload the page.</p>
                    </div>
                );

                const links = value.folders.map(folder =>
                    <NavLink
                        key={folder.id}
                        className='folder-link'
                        activeClassName="highlighted"
                        to={`/folder/${folder.id}`}
                    >
                        {folder.name}
                    </NavLink>
                );

                return (
                    <nav className="sidebar">
                        {links}
                        {error ? errorHTML : ''}
                        <NavLink
                            className='folder-link'
                            activeClassName='highlighted'
                            to={'/add-folder'}
                        >
                            Add folder
                        </NavLink>
                    </nav>
                )
            }}
        </NotesContext.Consumer>
    );
}

export default SidebarLinks;