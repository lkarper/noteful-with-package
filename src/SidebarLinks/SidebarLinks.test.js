import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import SidebarLinks from './SidebarLinks';

describe('SidebarLinks component', () => {

    it('renders the UI without crashing', () => {
        const wrapper = mount(
            <BrowserRouter>
                <SidebarLinks />
            </BrowserRouter>
        );
        wrapper.unmount();
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <SidebarLinks />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });

    it('renders an error message when an error is passed in context', () => {
        const wrapper = mount(
            <BrowserRouter>
                <SidebarLinks />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });

});