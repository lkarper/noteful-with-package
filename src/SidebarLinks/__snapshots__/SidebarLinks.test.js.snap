// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarLinks component renders an error message when an error is passed in context 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SidebarLinks>
      <nav
        className="sidebar"
      >
        <div
          className="folder-load-error"
        >
          <h2>
            Sorry, could not load folders from the server: 
            Error loading folders
          </h2>
          <p>
            Check your network connection and reload the page.
          </p>
        </div>
        <NavLink
          activeClassName="highlighted"
          className="folder-link"
          to="/add-folder"
        >
          <Link
            aria-current={null}
            className="folder-link"
            to={
              Object {
                "hash": "",
                "pathname": "/add-folder",
                "search": "",
                "state": null,
              }
            }
          >
            <LinkAnchor
              aria-current={null}
              className="folder-link"
              href="/add-folder"
              navigate={[Function]}
            >
              <a
                aria-current={null}
                className="folder-link"
                href="/add-folder"
                onClick={[Function]}
              >
                Add folder
              </a>
            </LinkAnchor>
          </Link>
        </NavLink>
      </nav>
    </SidebarLinks>
  </Router>
</BrowserRouter>
`;

exports[`SidebarLinks component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SidebarLinks>
      <nav
        className="sidebar"
      >
        <NavLink
          activeClassName="highlighted"
          className="folder-link"
          key="b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1"
          to="/folder/b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1"
        >
          <Link
            aria-current={null}
            className="folder-link"
            to={
              Object {
                "hash": "",
                "pathname": "/folder/b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
                "search": "",
                "state": null,
              }
            }
          >
            <LinkAnchor
              aria-current={null}
              className="folder-link"
              href="/folder/b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1"
              navigate={[Function]}
            >
              <a
                aria-current={null}
                className="folder-link"
                href="/folder/b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1"
                onClick={[Function]}
              >
                Important
              </a>
            </LinkAnchor>
          </Link>
        </NavLink>
        <NavLink
          activeClassName="highlighted"
          className="folder-link"
          key="b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1"
          to="/folder/b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1"
        >
          <Link
            aria-current={null}
            className="folder-link"
            to={
              Object {
                "hash": "",
                "pathname": "/folder/b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
                "search": "",
                "state": null,
              }
            }
          >
            <LinkAnchor
              aria-current={null}
              className="folder-link"
              href="/folder/b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1"
              navigate={[Function]}
            >
              <a
                aria-current={null}
                className="folder-link"
                href="/folder/b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1"
                onClick={[Function]}
              >
                Super
              </a>
            </LinkAnchor>
          </Link>
        </NavLink>
        <NavLink
          activeClassName="highlighted"
          className="folder-link"
          key="b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1"
          to="/folder/b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1"
        >
          <Link
            aria-current={null}
            className="folder-link"
            to={
              Object {
                "hash": "",
                "pathname": "/folder/b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1",
                "search": "",
                "state": null,
              }
            }
          >
            <LinkAnchor
              aria-current={null}
              className="folder-link"
              href="/folder/b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1"
              navigate={[Function]}
            >
              <a
                aria-current={null}
                className="folder-link"
                href="/folder/b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1"
                onClick={[Function]}
              >
                Spangley
              </a>
            </LinkAnchor>
          </Link>
        </NavLink>
        <NavLink
          activeClassName="highlighted"
          className="folder-link"
          to="/add-folder"
        >
          <Link
            aria-current={null}
            className="folder-link"
            to={
              Object {
                "hash": "",
                "pathname": "/add-folder",
                "search": "",
                "state": null,
              }
            }
          >
            <LinkAnchor
              aria-current={null}
              className="folder-link"
              href="/add-folder"
              navigate={[Function]}
            >
              <a
                aria-current={null}
                className="folder-link"
                href="/add-folder"
                onClick={[Function]}
              >
                Add folder
              </a>
            </LinkAnchor>
          </Link>
        </NavLink>
      </nav>
    </SidebarLinks>
  </Router>
</BrowserRouter>
`;
